runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SdkComponents.java

```diff
@@ -239,10 +273,10 @@ public String getOnlyEnvironmentId() {
   }
 
   private String uniqify(String baseName, Set<String> existing) {
-    String name = baseName;
+    String name = namespace + baseName;
```

Comment:
Unless it is hideously expensive, it might be smart to choose a canonical meta-format now (that there are three components) and escape the components. For example slash-separated with slashes internal to the name escaped. This is mostly a reaction to more strings being concatenated in a way that is starting to look like it should be real structure.